orbit/pkg/update/swift_dialog.go

```diff
@@ -56,14 +60,16 @@ func (s *SwiftDialogDownloader) Run(cfg *fleet.OrbitConfig) error {
 			s.UpdateRunner.updater.RemoveTargetInfo("swiftDialog")
 			return err
 		}
+	}
 
-		if cfg.Notifications.RunSetupExperience {
-			// Then update immediately, since we need to get swiftDialog quickly to show the setup
-			// experience
-			_, err := s.UpdateRunner.UpdateAction()
-			if err != nil {
-				return err
-			}
+	// If we're running setup experience and we have the hashes we need to make sure we trigger
+	// an immediate update to get SwiftDialog installed and usable.
+	if cfg.Notifications.RunSetupExperience && !s.triggeredSetupExperienceUpdate {
+		s.triggeredSetupExperienceUpdate = true
```

Comment:
One more thing, does `s.UpdateRunner.UpdateAction` need to run only once? In main there's just `if cfg.Notifications.RunSetupExperience {`.

PS: If there's nothing to update then `s.UpdateRunner.UpdateAction` does nothing.

Reply:
Now that I think about it, if we do that then we'd be checking for updates every 30 seconds instead of every 15 minutes. Let's leave as is.